when linking fails, the command should not return SUCCESS

When the link command fails and returns false, the build is still considered successful.

/usr/bin/ld: cannot find -ljvm
collect2: error: ld returned 1 exit status
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS

--- a/src/main/java/com/gluonhq/NativeLinkMojo.java
+++ b/src/main/java/com/gluonhq/NativeLinkMojo.java
@@ -51,7 +51,8 @@ public class NativeLinkMojo extends NativeBaseMojo {
             getLog().debug("Start linking in " + tmpPath.toString());
 
             SubstrateDispatcher dispatcher = new SubstrateDispatcher(client, clientConfig);
-            dispatcher.nativeLink(getProjectClasspath());
+            boolean result = dispatcher.nativeLink(getProjectClasspath());
+            if (!result) throw new RuntimeException("Linking failed");
         } catch (Exception e) {
             e.printStackTrace();
             throw new MojoExecutionException("Error", e);

